content/src/service/synchronization/EventDeployer.ts

```diff
@@ -18,45 +18,18 @@ export class EventDeployer {
 
     constructor(private readonly cluster: ContentCluster,
         private readonly service: ClusterDeploymentsService) {
-            this.eventProcessor = new EventStreamProcessor((event, source) => this.wrapDeployment(this.prepareDeployment(event, source)))
-        }
-
-    processAllDeployments(allDeployments: Deployment[][], options?: HistoryDeploymentOptions) {
-        // Remove duplicates
-        const map: Map<EntityId, Deployment> = new Map()
-        allDeployments.forEach(serverDeployments => serverDeployments.forEach(deployment => map.set(deployment.entityId, deployment)))
-
-        // Unify
-        const unifiedDeployments = Array.from(map.values())
-
-        // Deploy
-        return this.deployDeployments(unifiedDeployments, options)
-    }
-
-    private async deployDeployments(deployments: Deployment[], options?: HistoryDeploymentOptions) {
-        // Determine whether I already know the entities
-        const entitiesInDeployments: EntityId[] = deployments.map(({ entityId }) => entityId)
-        const deployInfo = await this.service.areEntitiesAlreadyDeployed(entitiesInDeployments)
-        const newEntities: Set<EntityId> = new Set(Array.from(deployInfo.entries())
-            .filter(([, deployed]) => !deployed)
-            .map(([entityId]) => entityId))
-
-        // Keep only new deployments
-        const newDeployments = deployments.filter(event => newEntities.has(event.entityId));
-
-        if (deployments.length > 0) {
-            EventDeployer.LOGGER.debug(`History had ${deployments.length} entities, only ${newDeployments.length} new.`)
-            if (newDeployments.length > 0) {
-                EventDeployer.LOGGER.debug(`Will start to deploy the ${newDeployments.length} new entities.`)
-            }
+            this.eventProcessor = new EventStreamProcessor(
+                (entityIds) => this.service.areEntitiesAlreadyDeployed(entityIds),
+                (event, source) => this.wrapDeployment(this.prepareDeployment(event, source)))
         }
 
+    async processAllDeployments(deployments: Readable[], options?: HistoryDeploymentOptions) {
         // Process history and deploy it
-        return this.eventProcessor.processDeployments(newDeployments, options)
+        return this.eventProcessor.processDeployments(deployments, options)
     }
 
     /** Download and prepare everything necessary to deploy an entity */
-    private async prepareDeployment(deployment: Deployment, source?: ContentServerClient): Promise<DeploymentExecution> {
+    private async prepareDeployment(deployment: DeploymentWithAuditInfo, source?: ContentServerClient): Promise<DeploymentExecution> {
```

Comment:
Changed this because we are not getting the full deployment now, we are only getting the base properties and the audit info